bundles/specmate-connectors/src/com/specmate/connectors/internal/ConnectorUtil.java

```diff
@@ -0,0 +1,304 @@
+package com.specmate.connectors.internal;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.emf.common.util.TreeIterator;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.osgi.service.log.LogService;
+
+import com.specmate.common.exception.SpecmateException;
+import com.specmate.connectors.api.IRequirementsSource;
+import com.specmate.model.base.BaseFactory;
+import com.specmate.model.base.Folder;
+import com.specmate.model.base.IContainer;
+import com.specmate.model.base.IContentElement;
+import com.specmate.model.requirements.Requirement;
+import com.specmate.model.support.util.SpecmateEcoreUtil;
+import com.specmate.persistency.IChange;
+import com.specmate.persistency.ITransaction;
+
+public class ConnectorUtil {
+
+	/** Lock, thus only one sync runs at the same time */
+	private final static Lock lock = new ReentrantLock();
+
+	/** How many retrieved requirements to process in one transaction */
+	public static final int BATCH_SIZE = 100;
+
+	/** Max size of field */
+	private static final int MAX_FIELD_LENGTH = 4000;
+
+	/** The logging service */
+	private LogService logService;
+
+	/** The transaction */
+	private ITransaction transaction;
+
+	/** List of registered requirements sources */
+	private List<IRequirementsSource> requirementsSources;
+
+	public ConnectorUtil(List<IRequirementsSource> requirementsSources, ITransaction transaction,
```

Comment:
I find it strange that a class named "util" needs to be instantiated. In a util class i expect only static methods. I suggest to rename the class or (maybe better) to remove the constructor paramters and move them as parameters into the methods?